src/main/java/mate/academy/spring/controller/UserController.java

```
@@ -0,0 +1,30 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.UserResponseDto;
+import mate.academy.spring.service.UserService;
+import mate.academy.spring.service.dto.mapping.DtoResponseMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    private final UserService userService;
+    private final DtoResponseMapper<UserResponseDto, User> userResponseMapper;
+
+    @Autowired
+    public UserController(UserService userService,
+                          DtoResponseMapper<UserResponseDto, User> userResponseMapper) {
+        this.userService = userService;
+        this.userResponseMapper = userResponseMapper;
+    }
+
+    @GetMapping("/by-email")
+    public UserResponseDto getByEmail(@RequestParam String email) {
+        return userResponseMapper.toDto(userService.findByEmail(email).get());
```

let's throw an exception in case of null in optional